packages/api/src/routes/middlewares/auth.ts

```diff
@@ -9,49 +10,63 @@ import { getCxIdOrFail } from "../util";
  * Process the API key and get the customer id.
  * The customer id is stored on the Request, property 'cxId'.
  */
-export const processAPIKey = (req: Request, res: Response, next: NextFunction): void => {
+export function processCxId(req: Request, res: Response, next: NextFunction): void {
   try {
     // Just gets the cxId from the API Key, the actual auth is done on API GW.
     // Downstream routes should check whether `cxId` is present on the request or not.
     const encodedApiKey = req.header("x-api-key");
     req.cxId = getCxIdFromApiKey(encodedApiKey);
   } catch (error) {
-    // noop - auth is done on API GW level, this is just to make data available downstream
+    try {
+      req.cxId = getCxIdFromReq(req);
+    } catch (error) {
+      // noop - auth is done on API GW level, this is just to make data available downstream
+    }
   }
```

Comment:
The main change: tries to get the cxId from the request if API not set. Obviously, auth is done on API GW and here we're just getting the customer ID from the request.